cudapoa/include/claragenomics/cudapoa/batch.hpp

```diff
@@ -44,6 +45,58 @@ struct Entry
 /// A type defining the set and order of Entry's in which a POA is processed.
 typedef std::vector<Entry> Group;
 
+/// A structure to hold  upper limits for data size processed in POA batches
+struct BatchSize
+{
+    /// Maximum number of elements in a sequence
+    uint32_t max_sequence_size;
```

Comment:
there's one thing I forgot to mention @r-mafi - earlier in our development we decided to use `int32_t` for integer data types, and then use the `throw_on_negative` from https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/dev-v0.5.0/common/utils/include/claragenomics/utils/signed_integer_utils.hpp#L39 to catch negative numbers. this helps catch cases where use inputs a negative numbers to the API with type as `uint32_t`

Reply:
e.g. https://github.com/clara-genomics/ClaraGenomicsAnalysis/pull/327/files#diff-c4fc2db9ecbc3d4544f4422682e91668R59

Reply:
sure, changed to int32_t. This may require updating the python binding again.